lib/rust/ensogl/core/src/display/scene/layer.rs

```diff
@@ -868,6 +875,7 @@ impl SublayersModel {
     fn remove(&mut self, layer_id: LayerId) {
         if let Some(ix) = self.layer_ix(layer_id) {
             self.layers.remove(ix);
+            self.layer_placement.remove(&layer_id);
```

Comment:
What the lack of this line was causing?

Reply:
Adding a line to "look consistent" is IMO a bad way of code modification. Please double check if it's needed and either remove it being sure it is not needed, or keep it and then write explanation what exactly lack of this line was causing, please.

Reply:
Ok, I removed this line. The lack of this line obviously causing a memory leak, but the actual effect is insignificant in real-world conditions. This line is not needed, as it neither fixes an immediate bug nor is necessary for scroll area to work.
1. I say "obviously", because we always need to remove entries from both collections in `remove_*` if we add to both of them in `add_*`. See [my comment](https://github.com/enso-org/enso/pull/3384#discussion_r854422083). Failing to do so leads to a constantly growing `layer_placement` hashmap.
2. I say "insignificant in real-world conditions", because the actual size of a leak is neglectible. It's hard to calculate the precise numbers because I don't know the exact memory layout of the hashmap, but we can estimate that each `HashMap<K, V>` entry takes at least `size_of::<K>() + size_of::<V>()` bytes, so `8 + 8 = 16` bytes in our case. Adding and removing 1'000'000 sublayers per frame would cause a 16 Mb/frame leak. However, I doubt we would ever create such a large amount of *unique* sublayers.